### ISSUE TITLE
if 'only-if-cached' cache control value is set, it should return 504 not 502ader 

### ISSUE THREAD

**Issue by huewu** [candrews/HttpResponseCache#5]
first of all, 
thx for this great library. it really help me. 

when only-if-cached header is set,  below response is responsed,

```
private static final CacheResponse BAD_GATEWAY_RESPONSE = new CacheResponse() {
    @Override public Map<String, List<String>> getHeaders() throws IOException {
        Map<String, List<String>> result = new HashMap<String, List<String>>();
        result.put(null, Collections.singletonList("HTTP/1.1 502 Bad Gateway"));
        return result;
    }
```

but, rfc 2616 sys, it should be HTTP/1.1 504 Gateway Timeout
...
To do this, the client may include the only-if-cached directive in a request. If it receives this directive, a cache SHOULD either respond using a cached entry that is consistent with the other constraints of the request, or respond with a 504 (Gateway Timeout) status.
...

please check this doc.
http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.9.4

thx again.

**Comment by candrews** [comment]
Reported in the Android bug tracker: https://code.google.com/p/android/issues/detail?id=28294

**Comment by candrews** [comment]
Thanks! Included in ec2b98aed81af1145ebad527fe96bbaae323de0c
